spec/services/force/soql/preference_service_spec.rb

```diff
@@ -23,4 +23,35 @@
       end
     end
   end
+
+  describe 'buildAppPreferencesSearch' do
+    it 'creates the expected query string for a single search term' do
+      VCR.use_cassette('services/soql/preference_service') do
+        filters = subject.buildAppPreferencesSearch('search_string')
+        expected_filters = [
+          "Application__r.Name like '%search_string%' OR ",
+          "Application__r.Applicant__r.First_Name__c like '%search_string%' OR ",
+          "Application__r.Applicant__r.Last_Name__c like '%search_string%' OR ",
+          "Application__r.Applicant__r.Email__c like '%search_string%'"
+        ].join('')
+        expect(filters). to eq(expected_filters)
+      end
+    end
+    it 'creates the expected query string for multiple search terms' do
+      VCR.use_cassette('services/soql/preference_service') do
+        filters = subject.buildAppPreferencesSearch('string1,string2')
+        expected_filters = [
+          "Application__r.Name like '%string1%' OR ",
+          "Application__r.Applicant__r.First_Name__c like '%string1%' OR ",
```

Comment:
the %'s here are wildcards right? So we're checking if the string is contained, not exact equality?

Reply:
correct, in the actual query, not in the test though.